reactive-banana/src/Reactive/Banana/Prim/Combinators.hs

```diff
@@ -134,9 +133,10 @@ switchP pp = mdo
             return Nothing
         -- fetch value from old parent
         eval = readPulseP =<< readLatchP lp
-    
+
     p1 <- newPulse "switchP_in" switch :: Build (Pulse ())
     p1 `dependOn` pp
     p2 <- newPulse "switchP_out" eval
+    p2 `dependOn` p
```

Comment:
Before, `p2` was initialized with no parents, and would not have a parent until `pp` fired. `p1`, a child of `pp`, would switch `p2`'s parent to the newly fired event.

Now, `p2` does have an initial parent - `p`. This, like before, will get switched away to a new parent by `p1` when `pp` fires.